sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+
+#nullable enable
+
+namespace Azure.Core.Pipeline
+{
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
+    internal class DiagnosticScopeFactory
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
+    {
+        private readonly string? _resourceProviderNamespace;
+        private readonly DiagnosticListener? _source;
+
+        public DiagnosticScopeFactory(string clientNamespace, string? resourceProviderNamespace, bool isActivityEnabled)
+        {
+            _resourceProviderNamespace = resourceProviderNamespace;
+            IsActivityEnabled = isActivityEnabled;
+            if (IsActivityEnabled)
+            {
+                _source = new DiagnosticListener(clientNamespace);
+            }
+        }
+
+        public bool IsActivityEnabled { get; }
+
+        public DiagnosticScope CreateScope(string name)
```

Comment:
Just for my own curiosity, why are we pulling this out into a base class?

Reply:
Because EH uses DiagnosticScopes but doesn't care about HTTP stuff.